tests/stdlib/tmarshal.nim

```diff
@@ -105,3 +105,16 @@ r = to[Something](data2)
 
 echo r.x, " ", r.y
 
+
+block:
+  type Foo = object
+    a1: string
+    a2: string
+    a3: seq[string]
+    a4: seq[int]
+    a5: seq[int]
+    a6: seq[int]
+  var foo = Foo(a2:"", a4: @[], a6: @[1])
+  foo.a6.setLen 0
+  # Note the difference between a5: null and a6: []
```

Comment:
Yeah... There shouldn't be any.

Reply:
ok, can we merge this first and address this point in a future PR? this PR adds to breaking change (only affects string but with no change in serialization), and fixing seq requires a (maybe innocuous but who knows) breaking  change in case someone deserializes from a saved json string

Reply:
/cc @Araq ping

Reply:
Why is ``tySequence`` still listed in ``isNil``?

Reply:
that's what I had initially attempted in https://github.com/nim-lang/Nim/pull/8934 ; as explained in https://github.com/nim-lang/Nim/pull/8942#issue-214437510 it was causing other issues; so I'd rather get this PR merged first (which addresses  partially #8916, addressing the string case) and leave the `tySequence` case for a subsequent PR